ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java

```diff
@@ -364,13 +368,24 @@ public List<PartitionFileBaseInfo> getSnapshotDetail(AmoroTable<?> amoroTable, l
                         commitId,
                         DataFileType.ofContentId(f.content().id()),
                         snapshotTime,
-                        arcticTable.spec().partitionToPath(f.partition()),
+                        getPartitionSpecBySpecId(arcticTable, f.specId())
+                            .partitionToPath(f.partition()),
                         f.path().toString(),
                         f.fileSizeInBytes(),
                         "remove")));
     return result;
   }
 
+  private PartitionSpec getPartitionSpecBySpecId(ArcticTable arcticTable, Integer fileSpecId) {
```

Comment:
You can reuse `com.netease.arctic.utils.ArcticTableUtil#getArcticTablePartitionSpecById`.

Reply:
> You can reuse `com.netease.arctic.utils.ArcticTableUtil#getArcticTablePartitionSpecById`.

Agree